arch/xtensa/src/esp32s3/esp32s3_sdmmc.c

```diff
@@ -0,0 +1,2924 @@
+        }
+    }
+
+  /* No this is a new address or value */
+
+  else
+    {
+      /* Did we print "..." for the previous value? */
+
+      if (count > 3)
+        {
+          /* Yes.. then show how many times the value repeated */
+
+          mcerr("%s: [repeats %d more times]\n", func, count - 3);
+        }
+
+      /* Save the new address, value, and count */
+
+      prevaddr = addr;
+      preval   = val;
+      count    = 1;
+    }
+
+  /* Show the register value read */
+
+  mcerr("%s: %08x->%08x\n", func, addr, val);
+  return val;
+}
+#endif
+
+/****************************************************************************
+ * Name: esp32s3_putreg
+ *
+ * Description:
+ *   This function may to used to intercept an monitor all register accesses.
+ *   Clearly this is nothing you would want to do unless you are debugging
+ *   this driver.
+ *
+ * Input Parameters:
+ *   val - The value to write to the register
+ *   addr - The register address to read
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_ESP32S3_SDMMC_REGDEBUG
+static void __esp32s3_putreg(const char *func, uint32_t val, uint32_t addr)
+{
+  /* Show the register value being written */
+
+  mcerr("%s: %08x<-%08x\n", func, addr, val);
+
+  /* Write the value */
+
+  putreg32(val, addr);
+}
+#endif
+
+/****************************************************************************
+ * Name: esp32s3_ciu_sendcmd
+ *
+ * Description:
+ *   Function to send command to Card interface unit (CIU)
+ *
+ * Input Parameters:
+ *   cmd - The command to be executed
+ *   arg - The argument to use with the command.
+ *
+ * Returned Value:
+ *   Returns zero on success.  One will be returned on a timeout.
+ *
+ ****************************************************************************/
+
+static int esp32s3_ciu_sendcmd(uint32_t cmd, uint32_t arg)
+{
+  clock_t watchtime;
+
+  DEBUGASSERT((esp32s3_getreg(ESP32S3_SDMMC_CMD) & SDMMC_CMD_STARTCMD) == 0);
+
+  watchtime = clock_systime_ticks();
+
+  while ((esp32s3_getreg(ESP32S3_SDMMC_CMD) & SDMMC_CMD_STARTCMD) != 0)
+    {
+      if (watchtime - clock_systime_ticks() > SDCARD_CMDTIMEOUT)
+        {
+          mcerr("TMO Timed out (%08X)\n",
+                esp32s3_getreg(ESP32S3_SDMMC_CMD));
+          return 1;
+        }
+    }
+
+  /* Set command arg reg */
+
+  cmd |= SDMMC_CMD_STARTCMD | SDMMC_CMD_USE_HOLE;
+
+  esp32s3_putreg(arg, ESP32S3_SDMMC_CMDARG);
+  esp32s3_putreg(cmd, ESP32S3_SDMMC_CMD);
+
+  mcinfo("cmd=0x%x arg=0x%x\n", cmd, arg);
+
+  /* Poll until command is accepted by the CIU, or we timeout */
+
+  return 0;
+}
+
+static void configure_pin(uint8_t gpio_pin, uint8_t sdio_pin,
+                          gpio_pinattr_t attr)
+{
+  esp32s3_configgpio(gpio_pin, attr);
+
+  if (attr & INPUT)
+    esp32s3_gpio_matrix_in(gpio_pin, sdio_pin, false);
+
+  if (attr & OUTPUT)
+    esp32s3_gpio_matrix_out(gpio_pin, sdio_pin, false, false);
```

Comment:
According to [NuttX's C Coding Standard](https://nuttx.apache.org/docs/latest/contributing/coding_style.html#one-statement-per-line):

> One statement per line. There should never be more than one statement on a line.

```suggestion
  esp32s3_configgpio(gpio_pin, attr);

    if (attr & INPUT)
    {
      esp32s3_gpio_matrix_in(gpio_pin, sdio_pin, false);
    }

  if (attr & OUTPUT)
    {
      esp32s3_gpio_matrix_out(gpio_pin, sdio_pin, false, false);
    }
```